pkg/monitoring/WampStorage.php

```diff
@@ -0,0 +1,205 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Enqueue\Monitoring;
+
+use Enqueue\Dsn\Dsn;
+use Thruway\ClientSession;
+use Thruway\Peer\Client;
+use Thruway\Transport\PawlTransportProvider;
+
+class WampStorage implements StatsStorage
+{
+    /**
+     * @var array
+     */
+    private $config;
+
+    /**
+     * @var Client
+     */
+    private $client;
+
+    /**
+     * @var Serializer
+     */
+    private $serialiser;
+
+    /**
+     * @var ClientSession
+     */
+    private $session;
+
+    /**
+     * @var Stats
+     */
+    private $stats;
+
+    /**
+     * The config could be an array, string DSN or null. In case of null it will attempt to connect to Thruway localhost.
+     *
+     * $config = [
+     *   'dsn'                 => 'wamp://127.0.0.1:9090',
+     *   'host'                => '127.0.0.1',
+     *   'port'                => '9090',
+     *   'topic'               => 'stats',
+     *   'max_retries'         => 15,
+     *   'initial_retry_delay' => 1.5,
+     *   'max_retry_delay'     => 300,
+     *   'retry_delay_growth'  => 1.5,
+     * ]
+     *
+     * or
+     *
+     * wamp://127.0.0.1:9090?max_retries=10
+     *
+     * @param array|string|null $config
+     */
+    public function __construct($config = 'wamp:')
+    {
+        if (empty($config)) {
```

Comment:
check that 3rd party library is installed. (add the lib to composer suggest section).